src/DistributedWriteAheadLog/KafkaWAL.cpp

```diff
@@ -137,9 +137,13 @@ std::shared_ptr<rd_kafka_topic_s> KafkaWAL::initProducerTopicHandle(const KafkaW
         /// std::make_pair("delivery.timeout.ms", std::to_string(kLocalMessageTimeout)),
         /// FIXME, partitioner
         std::make_pair("partitioner", "consistent_random"),
-        std::make_pair("compression.codec", "inherit"),
     };
 
+    if (walctx.enable_compress_internal_topic)
```

Comment:
Once we put this flag to `walctx`, it doesn't matter if it is internal topic or not since we lose the `internal` context. So renaming the flag name to something like `client_side_compression` seems more readable.